api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+            accountData.setRedhatAccountName(accountData.getRedhatAccountId());
+        }
+        AccountValidationResponse accountResponse = validateRequest(accountData);
+        if (accountResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info(VALIDATION_FAILED_DUE_TO_MISSING_PARAMETERS);
+            return accountResponse;
+        }
+        LOGGER.info(ADDING_ACCOUNT);
+        AccountValidationResponse validateResponse;
+        try {
+            validateResponse = createAccountInDb(accountData.getRedhatAccountId(), accountData.getRedhatAccountName(),
+                    Constants.REDHAT, accountData.getCreatedBy());
+
+            if (validateResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+                LOGGER.info(ACCOUNT_EXISTS);
+                return validateResponse;
+            }
+            BasicSessionCredentials credentials = credentialProvider.getBaseAccCredentials();
+            Regions region = Regions.fromName(System.getenv(REGION));
+            String roleName = System.getenv(PALADINCLOUD_RO);
+            AWSSecretsManager secretClient = AWSSecretsManagerClientBuilder.standard()
+                    .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                    .withRegion(region).build();
+
+            CreateSecretRequest createRequest = new CreateSecretRequest()
+                    .withName(secretManagerPrefix + FORWARD_SLASH + roleName + FORWARD_SLASH + serviceType() +
+                            FORWARD_SLASH + accountData.getRedhatAccountId())
+                    .withSecretString(String.format(REDHAT_TOKEN_TEMPLATE, accountData.getRedhatToken()));
+
+            CreateSecretResult createResponse = secretClient.createSecret(createRequest);
+            LOGGER.info(SECRET_RESPONSE, createResponse);
+            //update redhat enable flag for scheduler job
+            updateConfigProperty(REDHAT_ENABLED, TRUE, JOB_SCHEDULER);
+            validateResponse.setValidationStatus(SUCCESS);
+            validateResponse.setMessage(String.format(ADDING_ACCOUNT_SUCCESS_MESSAGE, accountData.getRedhatAccountId()));
+            validateResponse.setAccountName(accountData.getRedhatAccountName());
+        } catch (Exception e) {
+            deleteAccountFromDB(accountData.getRedhatAccountId());
+            validateResponse = new AccountValidationResponse();
+            validateResponse.setValidationStatus(FAILURE);
+            validateResponse.setMessage(String.format(ERROR_MESSAGE_TEMPLATE, CREATING));
+            validateResponse.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), CREATING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE, CREATING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), CREATING));
+        }
+        return validateResponse;
+    }
+
+    @Override
+    public AccountValidationResponse deleteAccount(String accountId) {
```

Comment:
Why is delete account returns AccountValidationResponse? Shouldn't it be simply RedhatResponse as an example that is common for all API calls.

Reply:
yes but I was using the deleteAccount from **interface AccountsService** its used for all the account types